pkg/cake/image_tree.go

```diff
@@ -159,8 +160,36 @@ func WalkBuildGraph(graph *Image, apply func(image *Image)) {
 		if len(queue) == 0 {
 			break
 		}
-		image := queue[len(queue)-1]
+		image := queue[0]
+		queue[0] = nil
+		queue = queue[1:]
 		apply(image)
-		queue = append(image.Children, queue[:len(queue)-1]...)
+		queue = append(queue, image.Children...)
+	}
+}
+
+func WalkBuildGraphParallel(graph *Image, apply func(image *Image)) {
```

Comment:
should we add a comment describing how this method works and how it is different from `WalkBuildGraph`?

Reply:
done